pkg/eval/evaltest/evaltest.go

```diff
@@ -35,6 +35,7 @@ type TestCase struct {
 }
 
 type Result struct {
+	ValueRe   bool // are ValueOut values treated as a regexp matching the corresponding value
```

Comment:
Support for regexp matching is better implemented using a special type (say `MatchingRegexp`), like how Approximately works. This will allow mixing outputs that should be regexp-matched and those that don't:

```
  That("some code").Puts("foo", MatchingRegexp{Pattern: `.*bar`})
```

Reply:
Good idea. This change doesn't require the generalization but it could be useful in the future and is a better API. Done.